### PR TITLE
add have their center in and contain center of  in spatial join

### PR DESCRIPTION
**Pull request by UrbanGISer**
add have their center in and contain center of  in spatial join

### CODE REVIEW

File: knime_extension/src/nodes/spatialtool.py [hunk 1]
```diff
@@ -299,9 +308,21 @@ def execute(self, exec_context: knext.ExecutionContext, left_input, right_input)
...
         right_gdf.to_crs(left_gdf.crs, inplace=True)
-        gdf = left_gdf.sjoin(
-            right_gdf, how=self.join_mode.lower(), predicate=self.match_mode.lower()
-        )
+        if self.match_mode.lower() not in ["have_their_center_in", "contain_center_of"]:
+            gdf = left_gdf.sjoin(
+                right_gdf, how=self.join_mode.lower(), predicate=self.match_mode.lower()
+            )
+        elif self.match_mode.lower() == "have_their_center_in":
+            left_gdf["rep_center"] = left_gdf.representative_point()
```
Comment (koettert) [inline knime_extension/src/nodes/spatialtool.py:316]: Make sure that this column name does not exist

File: knime_extension/src/nodes/spatialtool.py [hunk 2]
```diff
@@ -299,9 +308,21 @@ def execute(self, exec_context: knext.ExecutionContext, left_input, right_input)
...
+            left_gdf_temp = left_gdf.set_geometry("rep_center")
+            gdf = gp.sjoin(left_gdf_temp, right_gdf, predicate="within")
+            gdf = gdf.set_geometry(self.left_geo_col).drop(columns=["rep_center"])
+        else:
+            right_gdf["rep_center"] = right_gdf.representative_point()
```
Comment (koettert) [inline knime_extension/src/nodes/spatialtool.py:321]: Make sure that this column name does not exist

**Review by koettert — CHANGES_REQUESTED** [review]

File: knime_extension/src/nodes/spatialtool.py (hunk 1 above)
Comment (UrbanGISer) [reply knime_extension/src/nodes/spatialtool.py:316]: Use   get_unique_column_name to get the name

elif self.match_mode.lower() == "have_their_center_in":
            REP_Center= knut.get_unique_column_name("rep_center", left_input.schema) 
            left_gdf[REP_Center] = left_gdf.representative_point()
            left_gdf_temp = left_gdf.set_geometry(REP_Center)
            gdf = gp.sjoin(left_gdf_temp, right_gdf, predicate="within")
            gdf = gdf.set_geometry(self.left_geo_col).drop(columns=[REP_Center])
        else:
            REP_Center= knut.get_unique_column_name("rep_center", right_input.schema) 
            right_gdf[REP_Center] = right_gdf.representative_point()
            right_gdf_temp = right_gdf.set_geometry(REP_Center).drop(
                columns=[self.right_geo_col]
            )
            gdf = gp.sjoin(left_gdf, right_gdf_temp, predicate="contains")

File: knime_extension/src/nodes/spatialtool.py (hunk 2 above)
Comment (UrbanGISer) [reply knime_extension/src/nodes/spatialtool.py:321]: same as above

**Comment by UrbanGISer** [comment]
All revised accordingly, please help review that, thanks!

File: knime_extension/src/nodes/spatialtool.py [hunk 3]
```diff
@@ -308,20 +318,37 @@ def configure(self, configure_context, left_input_schema, right_input_schema):
...
+            rep_center = knut.get_unique_column_name("rep_center", left_input.schema)
+            left_gdf[rep_center] = left_gdf.representative_point()
+            left_gdf_temp = left_gdf.set_geometry(rep_center)
+            gdf = gp.sjoin(left_gdf_temp, right_gdf, predicate="within")
+            gdf = gdf.set_geometry(self.left_geo_col).drop(columns=[rep_center])
+        else:
+            rep_center = knut.get_unique_column_name("rep_center", right_input.schema)
+            right_gdf[rep_center] = right_gdf.representative_point()
+            right_gdf_temp = right_gdf.set_geometry(rep_center).drop(
+                columns=[self.right_geo_col]
```
Comment (koettert) [inline knime_extension/src/nodes/spatialtool.py:344]: why do you drop the right_geo_col here and not the rep_center column as above?

**Review by koettert — CHANGES_REQUESTED** [review]

**Pull request merged**